sw/device/silicon_creator/rom/rom.c

```diff
@@ -672,11 +672,27 @@ void rom_interrupt_handler(void) {
   OT_UNREACHABLE();
 }
 
+void rom_exception_handler(void) {
+  uint32_t mcause;
+  CSR_READ(CSR_REG_MCAUSE, &mcause);
+  ibex_exception_code_t exception_code =
+      (ibex_exception_code_t)(mcause & kIbexExceptionCodeMax);
+
+  if (launder32(exception_code) == kIbexExceptionCodeLoadAccessFault) {
```

Comment:
Can you add a comment saying that this is an "expected" fault when reading flash with invalid ECC, and we just return execution back to code that tried to perform read operation.

Reply:
The handling of the flash ECC error has been re-written as pure assembly.
- It's shorter and consumes fewer precious ROM bytes.
- It's faster.
- There are fewer jumps and no C setup/teardown.